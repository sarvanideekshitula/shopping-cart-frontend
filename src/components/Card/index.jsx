import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.module.css';
import addToCart from '../../utils/addToCart';
import removeFromCart from '../../utils/removeFromCart';

const Card = (props) => {
  const {
    id, name, price, quantity, link, selectedQuantity, testId,
  } = props;
  const [minusDisabled, setMinusDisabled] = useState(false);
  const [plusDisabled, setPlusDisabled] = useState(false);
  const [modifiedQuantity, setModifiedQuantity] = useState(selectedQuantity || 0);
  function handleButtonDisable(input) {
    if (input >= quantity) {
      setPlusDisabled(true);
      setMinusDisabled(false);
    } else if (input <= 0) {
      setMinusDisabled(true);
      setPlusDisabled(false);
    } else {
      setMinusDisabled(false);
      setPlusDisabled(false);
    }
  }
  useEffect(() => {
    handleButtonDisable(modifiedQuantity);
  }, []);
  return (
    <div className={styles.card} data-testId={testId}>
      <div className={styles.image}>
        <img src={link} alt="artist" />
      </div>
      <div>
        <p>{name}</p>
        {' '}
        <p>
          MRP:
          {price}
        </p>
      </div>
      <div className={styles.buttons}>
        <button type="button" className={styles.button} onClick={() => { removeFromCart(id); setModifiedQuantity(modifiedQuantity - 1); handleButtonDisable(modifiedQuantity - 1); }} disabled={minusDisabled}>-</button>
        <p>{modifiedQuantity}</p>
        <button type="button" className={styles.button} onClick={() => { addToCart(id); setModifiedQuantity(modifiedQuantity + 1); handleButtonDisable(modifiedQuantity + 1); }} disabled={plusDisabled}>+</button>
      </div>
    </div>
  );
};
Card.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
  link: PropTypes.string.isRequired,
  selectedQuantity: PropTypes.number.isRequired,
  testId: PropTypes.string.isRequired,
};

export default Card;
