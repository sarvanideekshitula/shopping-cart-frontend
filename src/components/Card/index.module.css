.card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    box-shadow: 5px 10px 18px #888888;
    transition: 0.3s;
    display: flex;
    background-color: white;
    height: 20rem;
    width: 15rem;
    margin-bottom: 2rem;
    justify-content: center;
  }
  img{
    max-width:12rem;
    max-height: 10rem;
    width: auto;
    height: auto;
    padding: 2px 2px 2px 2px;
    border: solid 1px black;
    border-radius: 0.5rem;
  }
  .buttons{
      display: flex;
      flex-direction: row;
      height: 2rem;
      width: 8rem;
      justify-content: space-around;
      align-items: center;
  }