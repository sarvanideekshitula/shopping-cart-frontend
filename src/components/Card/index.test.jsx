import React from 'react';
import { render } from '@testing-library/react';
import Card from './index';

describe('the Genre container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Card id={1} name="name" price={20} quantity={1} link="link" selectedQuantity={1} testId="test" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
