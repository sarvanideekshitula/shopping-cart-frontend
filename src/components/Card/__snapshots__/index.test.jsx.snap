// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Genre container should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="test"
  >
    <div>
      <img
        alt="artist"
        src="link"
      />
    </div>
    <div>
      <p>
        name
      </p>
       
      <p>
        MRP:20
      </p>
    </div>
    <div>
      <button
        type="button"
      >
        -
      </button>
      <p>
        1
      </p>
      <button
        disabled=""
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
